### PR TITLE
Replace JM link with infobox for archived workflows [BW-672]

### PR DESCRIPTION
**Pull request by cjllanwarne**
Tested locally by moving the "archive limit" up and down. Starting at 6 years so that this doesn't _actually_ show up yet.

See screenshots in ticket.

### CODE REVIEW

File: src/pages/workspaces/workspace/jobHistory/SubmissionDetails.js
Comment (evanbernstein) [inline src/pages/workspaces/workspace/jobHistory/SubmissionDetails.js]: The AC for this ticket had this bullet:
"comment everywhere we have the deletion delay constant in code, a pointer to all the other locations this constant exists in code"
I'd love to see a comment here that points to the other repositories that this constant exists as well as a comment in that code that points here.

**Review by rsasch — COMMENTED** [review]
I found the UX of this confusing — because the greyed out icons have a hover and are a lighter color, I wouldn't think to click on them, but clicking on them is the only way to get a more detailed message and a link to a support article.

File: src/components/PopupTrigger.js
Comment (zarsky-broad) [inline src/components/PopupTrigger.js]: I'm _extremely_ hesitant to make a change like this to the base structure of something that's used in multiple places, rather than make it an optional addition.

File: src/pages/workspaces/workspace/jobHistory/SubmissionDetails.js
Comment (zarsky-broad) [inline src/pages/workspaces/workspace/jobHistory/SubmissionDetails.js]: If there's going to be a tooltip, perhaps it should be something like "learn more". I'd recommend checking with UX, if you haven't already.

File: src/pages/workspaces/workspace/jobHistory/SubmissionDetails.js
Comment (zarsky-broad) [inline src/pages/workspaces/workspace/jobHistory/SubmissionDetails.js]: I don't love overriding the icon. The info icon means click for more information. These mean other things. However, the infobox here just gives more information.

File: src/components/PopupTrigger.js
Comment (cjllanwarne) [reply src/components/PopupTrigger.js]: Oops, this wasn't meant to make it this far (and didn't work anyway), I'll remove it

File: src/pages/workspaces/workspace/jobHistory/SubmissionDetails.js
Comment (cjllanwarne) [reply src/pages/workspaces/workspace/jobHistory/SubmissionDetails.js]: 👍  I've pinged Joy on whether this is a good idea

File: src/pages/workspaces/workspace/jobHistory/SubmissionDetails.js
Comment (zarsky-broad) [inline src/pages/workspaces/workspace/jobHistory/SubmissionDetails.js]: This is being used for a boolean, so it should be `isArchived`.

**Review by rsasch — COMMENTED** [review]
I would defer to the UX team, but I think, instead of adding an additional visual variant (i.e. grey-with-hover-but-clickable icons), we keep things simple and use an existing one, e.g. the blue "i" that appears on the "Workflow Submission" page.

File: src/pages/workspaces/workspace/jobHistory/SubmissionDetails.js
Comment (zarsky-broad) [inline src/pages/workspaces/workspace/jobHistory/SubmissionDetails.js]: You should add parentheses here so it's whether you want it to evaluate

```js
(lastChangedDate && Date.parse(lastChangedDate)) < dateYearsAgo
```
vs
```js
lastChangedDate && (Date.parse(lastChangedDate) < dateYearsAgo)
```

File: src/pages/workspaces/workspace/jobHistory/SubmissionDetails.js
Comment (zarsky-broad) [inline src/pages/workspaces/workspace/jobHistory/SubmissionDetails.js]: You should look in to https://date-fns.org/v2.21.3/docs/differenceInYears, we have this library already.

Comment (zarsky-broad) [reply src/pages/workspaces/workspace/jobHistory/SubmissionDetails.js]: You can find an example in `ServiceAlerts.js` of using `differenceInDays`, note that we import it from the `fp` submodule of date-fns

File: src/pages/workspaces/workspace/jobHistory/SubmissionDetails.js
Comment (zarsky-broad) [reply src/pages/workspaces/workspace/jobHistory/SubmissionDetails.js]: I think she's transitioned off of Terra, I believe Jerome is the UX person now.

File: src/pages/workspaces/workspace/jobHistory/SubmissionDetails.js
Comment (zarsky-broad) [inline src/pages/workspaces/workspace/jobHistory/SubmissionDetails.js]:
```suggestion
      div({ style: { padding: '0.5rem 0' } }, [`This workflow's details have been archived (> ${archiveLimitString} old).`]),
```

File: src/pages/workspaces/workspace/jobHistory/SubmissionDetails.js
Comment (zarsky-broad) [inline src/pages/workspaces/workspace/jobHistory/SubmissionDetails.js]: I think the icon usually goes after the link text.

File: src/pages/workspaces/workspace/jobHistory/SubmissionDetails.js
Comment (cjllanwarne) [reply src/pages/workspaces/workspace/jobHistory/SubmissionDetails.js]: I dipped a toe into using the original Info icon here (https://github.com/DataBiosphere/terra-ui/pull/2489) but the UX feedback was to prefer this original method. Let me know if you had a different alternative in mind?

File: src/pages/workspaces/workspace/jobHistory/SubmissionDetails.js
Comment (cjllanwarne) [reply src/pages/workspaces/workspace/jobHistory/SubmissionDetails.js]: No longer applies after switching to `differenceInDays` as suggested below

File: src/pages/workspaces/workspace/jobHistory/SubmissionDetails.js
Comment (cjllanwarne) [reply src/pages/workspaces/workspace/jobHistory/SubmissionDetails.js]: 👍  I ended up using `differenceInDays` because the `differenceInYears` was in _whole_ years, which isn't fine-grained enough for this info

File: src/pages/workspaces/workspace/jobHistory/SubmissionDetails.js
Comment (salonishah11) [inline src/pages/workspaces/workspace/jobHistory/SubmissionDetails.js]: Just a heads up, [this](https://github.com/broadinstitute/firecloud-develop/pull/2541) PR in firecloud-develop will set the archive-delay in prod to 1 year

**Comment by salonishah11** [comment]
It looks like there is still some discussion happening about grey-with-hover-and-clickable icons v/s tooltip-with-blue-icon that needs to checked with UX. Holding off reviewing until then.

File: src/pages/workspaces/workspace/jobHistory/SubmissionDetails.js
Comment (cjllanwarne) [reply src/pages/workspaces/workspace/jobHistory/SubmissionDetails.js]: See also https://github.com/broadinstitute/firecloud-develop/pull/2540. We need to keep them in sync whenever one is changed the other must be changed too

**Comment by cjllanwarne** [comment]
@salonishah11 @rsasch the feedback from Joy was to prefer this option over the info box

File: src/pages/workspaces/workspace/jobHistory/SubmissionDetails.js
Comment (rsasch) [inline src/pages/workspaces/workspace/jobHistory/SubmissionDetails.js]: Nit: include the words "support article" in the linked text
```suggestion
        'Please refer to the ',
        h(Link, {
          href: 'https://support.terra.bio/hc/en-us/articles/360060601631',
          ...Utils.newTabLinkProps
        }, ['Workflow Details Archived  support article', icon('pop-out', { size: 18 })]),
        ' for more details.'
```

**Review by rsasch — APPROVED** [review]
One small nit, otherwise 👍🏻

File: src/pages/workspaces/workspace/jobHistory/SubmissionDetails.js
Comment (salonishah11) [inline src/pages/workspaces/workspace/jobHistory/SubmissionDetails.js]: Based on the tech talk discussion, I think this should be `deletionLimitInYears` since the metadata would still be available on archiving, but not after it is moved to GCS. Same applies for places below: 'archive' => 'deletion' limit

File: src/pages/workspaces/workspace/jobHistory/SubmissionDetails.js
Comment (aednichols) [inline src/pages/workspaces/workspace/jobHistory/SubmissionDetails.js:129]: TOL, I think it flows a bit smoother without the parenthetical and symbolic greater-than
```suggestion
      div({ style: { padding: '0.5rem 0' } }, [`This workflow ran more than ${deletionDelayString} ago and is archived.`]),
```
Screen Shot 2021-05-13 at 12 54 32 PM

**Review by aednichols — APPROVED** [review]
Very slick and user friendly... 👍 I couldn't help myself and threw in a copy-editing TOL

File: src/pages/workspaces/workspace/jobHistory/SubmissionDetails.js
Comment (zarsky-broad) [inline src/pages/workspaces/workspace/jobHistory/SubmissionDetails.js]: Should this use `date-fns`'s `parseJson`?

**Review by zarsky-broad — APPROVED** [review]
Just one thought.

**Pull request merged**
